lib/phoenix_ecto/plugs/check_repo_ready.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Phoenix.Ecto.Plugs.CheckRepoReady do
+  @moduledoc """
+  A plug that does some checks on your application repos.
+
+  Checks if the storage is up (database is created) or if there are any pending migrations.
+  Both checks can raise an error if the conditions are not met.
+
+  ## Plug options
+
+  * `:otp_app` - name of the application which the repos are fetched from
+  * `:get_migration_function` - function that returns the migrations for a given `repo`.
+    Should return a list of tuples with three elements,
+    e.g. `[{state :: :up | :down, version :: integer(), name :: String.t()]]`
+    If `ecto_sql` dependecy is loaded, uses `Ecto.Migrator.migrations/1` by default.
+  """
+
+  @behaviour Plug
+
+  alias Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(%Conn{} = conn, opts) do
+    repos = Application.get_env(opts[:otp_app], :ecto_repos, [])
+
+    for repo <- repos, Process.whereis(repo) do
+      check_storage_up!(repo)
+      check_pending_migrations!(repo, opts)
+    end
+
+    conn
+  end
+
+  defp check_storage_up!(repo) do
+    adapter_implement_storage_status? =
+      Code.ensure_loaded?(repo.__adapter__) &&
+        function_exported?(repo.__adapter__, :storage_status, 1)
+
+    storage_status_down? = fn -> repo.__adapter__.storage_status(repo.config()) == :down end
+
+    if adapter_implement_storage_status? && storage_status_down?.(),
+      do: raise(Phoenix.Ecto.StorageNotCreatedError, repo: repo)
+  end
+
+  defp check_pending_migrations!(repo, opts) do
+    ecto_sql_loaded? = Code.ensure_loaded?(Ecto.Migrator)
+
+    # If the dependency `ecto_sql` is not loaded we can't check if
+    # there are pending migrations so we try to fail gracefully here
+    fallback_get_migrations_function =
+      if ecto_sql_loaded?, do: &Ecto.Migrator.migrations/1, else: fn _repo -> [] end
```

Comment:
I'm receiving a warning stating that `Ecto.Migrator` is not defined when `ecto_sql` is not loaded, (which makes sense). Do Do you folks know any way to go around this? Do we even want to suppress this warning? 👀 

![1](https://user-images.githubusercontent.com/14015177/68431114-2f609a80-0190-11ea-88d9-e3faa38812de.png)

Reply:
In your project config, do this: `[xref: [exclude: [Ecto.Migrator]]]` or similar. See `mix help xref`.

Reply:
Nice, TIL about `mix xref`. 🚀
Should I add this to phoenix_ecto's `mix.exs`? 🤔

Reply:
Yes.

Reply:
[fixed here](https://github.com/phoenixframework/phoenix_ecto/compare/846278a787c3300abfadce9a9d8097da9131afe2..5083d99f7f515143f5d00420d620f5491833597f)